firmware/controllers/algo/engine_configuration.cpp

```diff
@@ -1126,12 +1064,95 @@ void setDefaultConfiguration(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
+ */
+void setDefaultBoardConfiguration(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
+	// set initial pin groups
+	setDefaultBasePins(PASS_ENGINE_PARAMETER_SIGNATURE);
+
+	setDefaultSerialParameters(PASS_ENGINE_PARAMETER_SIGNATURE);
+
+	setCanDefaults(PASS_ENGINE_PARAMETER_SIGNATURE);
+
+	engineConfiguration->map.sensor.hwChannel = EFI_ADC_4;
+	engineConfiguration->clt.adcChannel = EFI_ADC_6;
+	engineConfiguration->iat.adcChannel = EFI_ADC_7;
+	engineConfiguration->afr.hwChannel = EFI_ADC_14;
+
+	engineConfiguration->accelerometerSpiDevice = SPI_DEVICE_1;
+	engineConfiguration->hip9011SpiDevice = SPI_DEVICE_2;
+	engineConfiguration->cj125SpiDevice = SPI_DEVICE_2;
+
+//	boardConfiguration->gps_rx_pin = GPIOB_7;
+//	boardConfiguration->gps_tx_pin = GPIOB_6;
+
+	boardConfiguration->triggerSimulatorPins[0] = GPIOD_1;
+	boardConfiguration->triggerSimulatorPins[1] = GPIOD_2;
+
+	boardConfiguration->triggerInputPins[0] = GPIOC_6;
+	boardConfiguration->triggerInputPins[1] = GPIOA_5;
+
+	//boardConfiguration->logicAnalyzerPins[1] = GPIOE_5; // GPIOE_5 is a popular option (if available)
+
+	boardConfiguration->boardTestModeJumperPin = GPIOB_0;
+
+	// set this to SPI_DEVICE_3 to enable stimulation
+	//boardConfiguration->digitalPotentiometerSpiDevice = SPI_DEVICE_3;
+	boardConfiguration->digitalPotentiometerChipSelect[0] = GPIOD_7;
+	boardConfiguration->digitalPotentiometerChipSelect[1] = GPIO_UNASSIGNED;
+	boardConfiguration->digitalPotentiometerChipSelect[2] = GPIOD_5;
+	boardConfiguration->digitalPotentiometerChipSelect[3] = GPIO_UNASSIGNED;
+
+	boardConfiguration->spi1mosiPin = GPIOB_5;
+	boardConfiguration->spi1misoPin = GPIOB_4;
+	boardConfiguration->spi1sckPin = GPIOB_3; // please note that this pin is also SWO/SWD - Single Wire debug Output
+
+	boardConfiguration->spi2mosiPin = GPIOB_15;
+	boardConfiguration->spi2misoPin = GPIOB_14;
+	boardConfiguration->spi2sckPin = GPIOB_13;
+
+	boardConfiguration->spi3mosiPin = GPIOB_5;
+	boardConfiguration->spi3misoPin = GPIOB_4;
+	boardConfiguration->spi3sckPin = GPIOB_3;
+	
+	// set optional subsystem configs
+#if EFI_MEMS
+	// this would override some values from above
+	configureAccelerometerPins(PASS_ENGINE_PARAMETER_SIGNATURE);
+#endif /* EFI_MEMS */
+
+#if EFI_HIP_9011
+	setHip9011FrankensoPinout();
+#endif /* EFI_HIP_9011 */
+
+#if EFI_FILE_LOGGING
+	setDefaultSdCardParameters(PASS_ENGINE_PARAMETER_SIGNATURE);
+#endif /* EFI_FILE_LOGGING */
+
+	boardConfiguration->is_enabled_spi_1 = false;
+	boardConfiguration->is_enabled_spi_2 = false;
+	boardConfiguration->is_enabled_spi_3 = true;
+}
+
 void resetConfigurationExt(Logging * logger, engine_type_e engineType DECLARE_ENGINE_PARAMETER_SUFFIX) {
 	enginePins.reset(); // that's mostly important for functional tests
 	/**
 	 * Let's apply global defaults first
 	 */
 	setDefaultConfiguration(PASS_ENGINE_PARAMETER_SIGNATURE);
+	/**
+	 * Let's apply global board defaults too, for most of the boards
+	 */
+	if (engineType != MINIMAL_PINS) {
```

Comment:
This is why we need all this refactoring! Not all engine configs require non-default hardware pins.